python/lsst/pipe/base/task.py

```diff
@@ -122,9 +122,10 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = self.ConfigClass()
             self._taskDict = dict()
-            if log is None:
-                log = pexLog.getDefaultLog()
-            self.log = pexLog.Log(log, self._fullName)
+            if log is not None and isinstance(log, Log):
+                self.log = Log.getLogger(log.getName() + '.' + self._fullName)
+            else:
+                self.log = Log.getLogger(self._fullName)
```

Comment:
No idea, but I'd still be in favor of reorganizing the logic to make it clear this is entirely about obtaining a name.